Makefile

```diff
@@ -99,6 +100,26 @@ $(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
 	stg repair
 	stg import -s ../patch/series
 
+	mkdir $(NON_UP_LOC)
+
+	if [ ! -z ${EXTERNAL_KERNEL_PATCHES} ];  then
+		wget ${EXTERNAL_KERNEL_PATCHES} -O patches.tar
+		tar -xf patches.tar -C $(NON_UP_LOC)
+	fi
+
+	# Precedence is given for external URL
+	if [ -z ${EXTERNAL_KERNEL_PATCHES} ] && [ x${INCLUDE_MLNX_PATCHES} == xy ]; then
```

Comment:
Thinking about the current kernel build, the kernel build is not platform-specific, and I'd rather not have platform-specific references in the build scripts. There's also a bit of a naming issue, where `EXTERNAL_KERNEL_PATCHES` could be set, but `INCLUDE_MLNX_PATCHES` could be `n`, and the patches would get applied, which seems counter-intuitive.

For the variable names, please don't have any platform-specific references within this file. Also, to make it a bit more clear, it should be something like this:
* If `INCLUDE_EXTERNAL_PATCHES` is not `y`, then nothing new happens; otherwise,
* If `EXTERNAL_KERNEL_PATCH_URL` is set, then the tarball is downloaded and extracted
* If `EXTERNAL_KERNEL_PATCH_TAR` is set, then the tarball is extracted.

Reply:
How about the minor change:
* If `EXTERNAL_KERNEL_PATCH_URL` is set by the user, then the tarball is downloaded and extracted
* If `INCLUDE_EXTERNAL_PATCHES=y` is set by the user the code uses `EXTERNAL_KERNEL_PATCH_TAR` which is supposed to be set in the platform/ dir and then the tarball is extracted using that absolute path. `EXTERNAL_KERNEL_PATCH_TAR` shouldn't be passed from user and must be hardcoded and is specific to the platform.

Reply:
That sounds good.

Reply:
Updated